Update Synology Drive Client to 3.0.1-12667

Keiishu opened the pull request:
- [X] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [X] Have you validated your manifest locally with `winget validate --manifest <path>`? 
- [X] Have you tested your manifest locally with `winget install --manifest <path>`?
- [X] Does your manifest conform to the [1.0 schema](https://github.com/microsoft/winget-cli/blob/master/doc/ManifestSpecv1.0.md)?

-----


###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/19672)

Keiishu committed:
Update Synology Drive Client to 3.0.1-12667

Keiishu committed:
Remove unused fields

OfficialEsco commented on manifests/s/Synology/DriveClient/3.0.1-12667/Synology.DriveClient.locale.en-US.yaml:
```diff
@@ -0,0 +1,26 @@
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.defaultlocale.1.0.0.schema.json
+
+PackageIdentifier: Synology.DriveClient
+PackageVersion: 3.0.1-12667
+PackageLocale: en-US
+Publisher: Synology Inc.
```
```suggestion
PackageVersion: 7.0.1.12667
PackageLocale: en-US
Publisher: Synology
```

OfficialEsco commented on manifests/s/Synology/DriveClient/3.0.1-12667/Synology.DriveClient.installer.yaml:
```diff
@@ -0,0 +1,20 @@
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.0.0.schema.json
+
+PackageIdentifier: Synology.DriveClient
+PackageVersion: 3.0.1-12667
```
```suggestion
PackageVersion: 7.0.1.12667
```

OfficialEsco commented on manifests/s/Synology/DriveClient/3.0.1-12667/Synology.DriveClient.yaml:
```diff
@@ -0,0 +1,8 @@
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.version.1.0.0.schema.json
+
+PackageIdentifier: Synology.DriveClient
+PackageVersion: 3.0.1-12667
```
```suggestion
PackageVersion: 7.0.1.12667
```

OfficialEsco reviewed (changes requested):
![image](https://user-images.githubusercontent.com/15158490/124394199-a308d100-dcfe-11eb-91d2-d15067d54e33.png)

Keiishu commented:
> 
> 
> ![image](https://user-images.githubusercontent.com/15158490/124394199-a308d100-dcfe-11eb-91d2-d15067d54e33.png)

Where does this come from? I only took the official Synology website as a reference, so I must say it is confusing.
The download link is for the version "3.0.1-12667", not "3.0.1-12667".
![image](https://user-images.githubusercontent.com/28761304/124394314-44902280-dcff-11eb-9f32-0f3ddf6e845a.png)

And the official company name is not referred as "Synology", but "Synology Inc.".
![image](https://user-images.githubusercontent.com/28761304/124394341-5b367980-dcff-11eb-851c-d1a59e134e7f.png)

Keiishu committed:
Fix Installer Validation issue

https://github.com/microsoft/winget-pkgs/pull/19672#pullrequestreview-698695035

Dennis14e commented:
Why was the version number changed from 3.0.1 to 7.0.1?
See https://www.synology.com/de-de/releaseNote/SynologyDriveClient

ItzLevvie commented:
cc @Dennis14e: Windows Package Manager requires the version to be from the Control Panel.

Please update your application's version number if it's incorrect.

OfficialEsco commented:
Ask the developers why they can't make a consistent version name.

Keiishu commented:
> 
> 
> Why was the version number changed from 3.0.1 to 7.0.1?
> See [synology.com/de-de/releaseNote/SynologyDriveClient](https://www.synology.com/de-de/releaseNote/SynologyDriveClient)

> 
> 
> ![image](https://user-images.githubusercontent.com/15158490/124394199-a308d100-dcfe-11eb-91d2-d15067d54e33.png)

Dennis14e commented:
Oh okay, that's weird from Synology.
Thanks for the clarification. @ItzLevvie @OfficialEsco @Keiishu
